manifests/00-cluster-role.yaml

```diff
@@ -136,13 +136,13 @@ rules:
   - create
 
 # Mirrored from assets/router/cluster-role.yaml
+# and expanded for assets/canary/route.yaml
 - apiGroups:
   - route.openshift.io
   resources:
   - routes
   verbs:
-  - list
-  - watch
+  - '*'
```

Comment:
I might be missing something, but how would the operator be able to take advantage of a new role + role binding in the `openshift-ingress-canary` namespace? The canary controller (which resides within the ingress operator) needs to be able to perform all (well, most at least, so maybe the "*" isn't necessary) actions on routes.

Reply:
The role would be in the `openshift-ingress-canary` namespace and allow full control over routes, and the role binding bind that role to the operator's service account.  Because it would be a role (not a clusterrole) in the `openshift-ingress-canary` namespace, the role binding should only grant access to routes within that namespace, if I understand correctly.  Something like the following should work (but I have not verified that it does):
```yaml
---
kind: Role
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: ingress-operator-canary-controller
  namespace: openshift-ingress-canary
  annotations:
    include.release.openshift.io/self-managed-high-availability: "true"
rules:
- apiGroups:
  - route.openshift.io
  resources:
  - routes
  verbs:
  - *
---
kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: ingress-operator-canary-controller
  namespace: openshift-ingress-canary
  annotations:
    include.release.openshift.io/self-managed-high-availability: "true"
subjects:
- kind: ServiceAccount
  name: ingress-operator
  namespace: openshift-ingress-operator
roleRef:
  kind: Role
  apiGroup: rbac.authorization.k8s.io
  name: ingress-operator-canary-controller
```

Reply:
Ill include these resources in the manifests in my next push and verify that they work. Thanks!

Reply:
@Miciah the CVO does not (and in my opinion, probably should not) create the `openshift-ingress-canary`, so the CVO cannot create a Role in the `openshift-ingress-canary` namespace. Additionally, the operator would also need similar route RBAC permissions to be able to create a role in the `openshift-ingress-canary` namespace for routes.

So I see 2 solutions:
1) The CVO creates the `openshift-ingress-canary` namespace, and then creates the ingress canary controller role.
(This seems like an awkward thing to do especially if the canary controller isnt a mandatory component of the operator)
2) We fall back to just giving the operator '*' permissions on routes in it's `cluster-role`.

Thoughts?

Reply:
Note, ive reverted this PR to use `*` permissions for routes in the operator's cluster role

Reply:
@Miciah https://kubernetes.io/docs/reference/access-authn-authz/rbac/#privilege-escalation-prevention-and-bootstrapping is the solution we are looking for! I amended this PR so that the ingress operator is granted `bind` and `escalate` permissions on `roles` and `rolebindings` in it's cluster role. I also added the `role.yaml` and `role-binding.yaml` assets for the ingress canary. With these new changes, the ingress operator's route permissions are extended to `*` for the `openshift-ingress-canary` namespace _only_. PTAL when you can.

Reply:
Going with route '*' permissions after a discussion with miciah. While these permissions are broad, this is the "least bad" approach. The above mentioned `escalate`/`bind` approach introduces far more concerning privilege escalation issues to the operator.